libs/mark-flow-ui/src/pages/contest_page.tsx

```diff
@@ -26,6 +33,9 @@ interface ContestPageProps {
   precinctId?: PrecinctId;
   updateVote: ContestProps['updateVote'];
   votes: VotesDict;
+  // Moves focus to the "Next" button after voter selects an option in a contest.
+  // This helps reduce repetitive action for accessibility device users.
+  moveFocusAfterUpdateVote?: boolean;
```

Comment:
I think it's a design decision - I thought it was visually distracting to apply the dotted border styles to the "Next" button as a touch screen user. But I could also see an argument for usability, since it does hint the user what to press next. Do you have a strong opinion?

Reply:
Ah that makes sense. I agree with you it's nice to only show the focus outlines if the user is using focus-based navigation.

That's making me think it might be nice to have two behavior groups:
- Assistive devices/keyboard nav
- Touch screen nav

If we can distinguish which of these cases triggered a button (which is possible using the event object), then we can trigger further focus behavior. I'm thinking that it's possible to switch between the various navigation modes even when the devices are connected. Example here: https://github.com/votingworks/vxsuite/blob/main/libs/ui/src/radio_group.tsx#L133-L144

I'm thinking this might be a nice approach to both make behavior consistent and simplify the architecture.

Reply:
Can you say more about what situations could be handled by checking `e.nativeEvent.pointerType`?

1. Focus `Next` or `Review` button after pressing button to make a contest vote
2. Blur `Next` button after selected
3. Something I missed?

1 is complicated because we want to change focus state on an element other than the one that was just pressed. We handle the contest button press pretty far down in ContestPage > Contest > {CandidateContest, YesNoContest, ...} > ContestChoiceButton. So we'd have to pass the event up from [`ContestChoiceButton.onPress()`](https://github.com/votingworks/vxsuite/blob/main/libs/ui/src/contest_choice_button.tsx#L79) to at least ContestPage, which is the component that's aware of the `Next` and `Review` buttons. Or we'd pass a function `(event: Event) => void` down from ContestPage to ContestChoiceButton.

2 is simpler because we just need to blur the focus on the same element that received the event, so we can handle similarly to `RadioGroup` and avoid passing an additional `onPress` handler to `LinkButton`.

I'm not sure case 1 becomes any simpler if we switch over to checking `nativeEvent.pointerType`, but probably worth doing case 2. Am I missing something that makes case 1 simpler?

Reply:
I was imagining that we'd just modify `Button` to pass the event to its `onPress` handler. Then the `handleUpdateVote` function could take a parameter derived from the button's event to determine whether to focus the Next button.

The reason I think this is an improvement is that then the frontend doesn't to query the backend to know if the PAT device is connected (at least, based on what I'm reading here - but I could be missing something). We'd be letting the browser event tell us what interaction method is being used, and removes the need to store and expose the PAT device status.

If that doesn't seem like an improvement though, I'm game to keep it as is.

Reply:
> The reason I think this is an improvement is that then the frontend doesn't to query the backend to know if the PAT device is connected 

That's true, and I think that simplification is worth the (IMO) messier function passing in the frontend. Let me confirm that moving focus around isn't an anti pattern for screen readers. Then I'll make these changes.